posthog/models/event/query_event_list.py

```diff
@@ -112,29 +114,75 @@ def query_events_list(
-                {"team_id": team.pk, "limit": limit, "offset": offset, **condition_params},
-                query_type="events_list",
-                workload=Workload.OFFLINE,
-            )
+    order = "DESC" if len(order_by) == 1 and order_by[0] == "-timestamp" else "ASC"
+    if prop_filters != "":
+        return insight_query_with_columns(
+            SELECT_EVENT_BY_TEAM_AND_CONDITIONS_FILTERS_SQL.format(
+                conditions=conditions, limit=limit_sql, filters=prop_filters, order=order
+            ),
+            {"team_id": team.pk, "limit": limit, "offset": offset, **condition_params, **prop_filter_params},
+            query_type="events_list",
+            workload=Workload.OFFLINE,
+        )
+    else:
+        return insight_query_with_columns(
+            SELECT_EVENT_BY_TEAM_AND_CONDITIONS_SQL.format(conditions=conditions, limit=limit_sql, order=order),
+            {"team_id": team.pk, "limit": limit, "offset": offset, **condition_params},
+            query_type="events_list",
+            workload=Workload.OFFLINE,
+        )
+
+
+def query_events_list_v2(
+    team: Team,
+    query: EventsQuery,
+) -> EventsQueryResponse:
+    # Note: This code is inefficient and problematic, see https://github.com/PostHog/posthog/issues/13485 for details.
+    # To isolate its impact from rest of the queries its queries are run on different nodes as part of "offline" workloads.
+    hogql_context = HogQLContext()
 
-    # events list v2 - hogql
+    limit = min(QUERY_MAXIMUM_LIMIT, QUERY_DEFAULT_LIMIT if query.limit is None else query.limit) + 1
+    offset = 0 if query.offset is None else query.offset
+    action_id = query.actionId
+    person_id = query.personId
+    order_by = query.orderBy
+    select = query.select
+    where = query.where
+    event = query.event
+
+    properties = []
+    properties.extend(query.fixedProperties or [])
+    properties.extend(query.properties or [])
+
+    limit_sql = "LIMIT %(limit)s"
+    if offset > 0:
+        limit_sql += " OFFSET %(offset)s"
+
+    conditions, condition_params = determine_event_conditions(
+        {
+            "before": (now() + timedelta(seconds=5)).isoformat(),
+            "person_id": person_id,
+            "event": event,
+        }
+    )
+    parsedProperties = [p["__root__"] if "__root__" in p else p for p in [p.dict() for p in properties]]
+    filter = Filter(team=team, data={"properties": parsedProperties}, hogql_context=hogql_context)
+    prop_filters, prop_filter_params = parse_prop_grouped_clauses(
+        team_id=team.pk, property_group=filter.property_groups, has_person_id_joined=False, hogql_context=hogql_context
+    )
+
+    if action_id:
+        try:
+            action = Action.objects.get(pk=action_id, team_id=team.pk)
+        except Action.DoesNotExist:
+            raise Exception("Action does not exist")
+        if action.steps.count() == 0:
+            raise Exception("Action does not have any steps")
```

Comment:
They're match groups rather than steps – `ActionStep` is to "match group" what `Team` is to "project"